components/802.15.4_RF/stm-s2lp-rf-driver/source/rf_configuration.h

```diff
@@ -33,6 +41,7 @@ void rf_conf_calculate_rx_filter_bandwidth_registers(uint32_t rx_bandwidth, uint
 void rf_conf_calculate_rssi_threshold_registers(int16_t rssi_threshold, uint8_t *rssi_th);
 uint32_t rf_conf_calculate_deviation(phy_modulation_index_e modulation_index, uint32_t datarate);
 int16_t rf_conf_cca_threshold_percent_to_rssi(uint8_t percent);
+uint8_t rf_conf_calculate_pa_level_dbm(double power_dbm);
```

Comment:
Don't use `double`. If you must use floating point, use `float`, as there's a higher chance of hardware support. Even then, do you really need floating point? It should be avoided if at all possible. I doubt you need it here. Indeed, you're passing a `uint32_t` to it, which seems unlikely, as you should be taking negative values.

Reply:
I've used double (or float) due to the fact that the power level from s2lp is in 0.5db increaments, but we can use uint32_t as well and force a 1db increament